sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala

```diff
@@ -471,6 +480,34 @@ class SessionCatalog(
   }
 
   /**
+   * Normalize TableIdentifier by consistently ensuring the following two rules.
+   * 1. System-side temporary tables should have None as database.
+   * 2. System-side temporary tables should have prefixed table names.
+   * Currently, only INFORMATION_SCHEMA tables are system-side temporary tables, and this function
+   * returns TableIdentifier("information_schema.databases", None).
+   */
+  protected def normalizeTableIdentifier(name: TableIdentifier): TableIdentifier = synchronized {
+    if (name.database.isDefined) {
+      if (name.database.contains(INFORMATION_SCHEMA_DATABASE)) {
+        TableIdentifier(s"$INFORMATION_SCHEMA_DATABASE.${name.table}", None)
+      } else {
+        name
+      }
+    } else {
+      val tableName = formatTableName(name.table)
+      if (tableName.startsWith(INFORMATION_SCHEMA_DATABASE + ".")) {
```

Comment:
Can you explain which cases will enter this processing? Is that possible we could hit backtick-quoted `INFORMATION_SCHEMA_DATABASE ` here?